platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponentActivationOptions.java

```diff
@@ -0,0 +1,209 @@
+package com.mapbox.mapboxsdk.location;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.android.core.location.LocationEngineRequest;
+import com.mapbox.mapboxsdk.maps.Style;
+
+/**
+ * A class which holds the various options for activating the Maps SDK's {@link LocationComponent} to eventually
+ * show the device location on the map.
+ */
+public class LocationComponentActivationOptions {
+
+  private final Context context;
+  private final Style style;
+  private final LocationEngine locationEngine;
+  private final LocationEngineRequest locationEngineRequest;
+  private final LocationComponentOptions locationComponentOptions;
+  private final int styleRes;
+  private final boolean useDefaultLocationEngine;
+
+  private LocationComponentActivationOptions(@NonNull Context context, @NonNull Style style,
+                                     @Nullable LocationEngine locationEngine,
+                                     @Nullable LocationEngineRequest locationEngineRequest,
+                                     @Nullable LocationComponentOptions locationComponentOptions,
+                                     int styleRes,
+                                     boolean useDefaultLocationEngine) {
+    this.context = context;
+    this.style = style;
+    this.locationEngine = locationEngine;
+    this.locationEngineRequest = locationEngineRequest;
+    this.locationComponentOptions = locationComponentOptions;
+    this.styleRes = styleRes;
+    this.useDefaultLocationEngine = useDefaultLocationEngine;
+  }
+
+  /**
+   * Convenience method to retrieve a {@link LocationComponentActivationOptions} object which is ready to build with
+   *
+   * @return a builder object
+   */
+  @NonNull
+  public static Builder builder(@NonNull Context context, @NonNull Style fullyLoadedMapStyle) {
+    return new LocationComponentActivationOptions.Builder(context, fullyLoadedMapStyle);
+  }
+
+  /**
+   * The application's current context
+   *
+   * @return the application's current context
+   */
+  @NonNull
+  public Context context() {
+    return context;
+  }
+
+  /**
+   * The proxy object for current map style. More info at {@link Style}
+   *
+   * @return the map's fully loaded Style object
+   */
+  @NonNull
+  public Style style() {
+    return style;
+  }
+
+  /**
+   * The {@link LocationEngine} which the {@link LocationComponent} should use
+   *
+   * @return the engine, or null if you'd like to only force location updates
```

Comment:
> or null if you'd like to only force location updates

I think this can be skipped because we can still initialize a default engine.

Reply:
Ok, now reads as 

```
* @return the engine
```